text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,618 @@
+
+This means, any field related component (i.e. Label, Input, ErrorMessage, etc.)
+would naturally inherit the state and all a11y props needed that is managed from
+the Headless-styles helpers (invalid, disabled, readOnly, etc.) and require no
+props of those from the user.
+
+This would also allow us to internally create error logs that would "make it hard
+to break something" which would inform the user when they are missing a FieldProvider.
+
+For a more complex component - see the [Detailed design](#detailed-design) section.
+
+## Motivation
+
+### Why are we doing this?
+
+There has been consistent requests for a React component library due to the nature
+of most developers either being either unaware of how to use a "headless" library
+or generally uncomfortable with being responsible for owning their own components.
+
+Having a component library that has a familiar interface which ultimately uses
+Headless-styles and React-utils under the hood should help break that barrier.
+
+### What use cases does it support?
+
+A component library would be most beneficial to teams/users that:
+
+1. Are mostly Backend developers that have little to no experience using Javascript or React.
+2. Junior frontend developers who have little to no experience in Javascript or React.
+3. Marketing departments who would like to use Pando, but have "just enough experience to be dangerous" in frontend development.
+4. Team leads that haven't been able to keep up with the technology trends but still need to code when possible.
+5. Teams that have the experience and understanding of Javascript and React, but don't have the time that may be required to use an API like Headless-styles (vs. dropping in an abstraction of it - component).
+
+### What is the expected outcome?
+
+The expected outcome of creating a component library should be:
+
+1. More adoption for Pando due to having a library that uses a familar pattern that people are generally more comfortable with.
+2. Should naturally teach users about the semantic token naming convention as it will inheritently use Headless-styles for prop definitions (which uses the same naming convention of the themes for
+consistency).
+3. Provide a developer experience that uses "React best practices" for a result that naturally allows scalability with custom designs via composable components, and high performance via advanced React patterns.
+4. Share the functionality as Lego blocks (create your own outcome based on what you combine/remove).
+5. By nature, it should increase the community contribution for the same reasons
+mentioned above (a familiar design).
+
+## Detailed design
+
+The design of the components library should first and foremost center around the
+Pando standards:
+
+1. A11y above all
+2. High Perfomance
+3. Easy to use
+
+Outside of these standards, there are 2 groups of components to consider:
+
+1. [Simple components](#simple-components) (Tag, Button, TextLink, etc.)
```

Comment:
(A-Team here again) Would it be possible to get a full list of simple & complex components as part of the RFC?

Reply:
@phoenixfire07 we shouldn't need to do that in the RFC based on the examples and is also a little outside of the scope of this RFC (i.e. that's more post-RFC work). However, I can dive deeper in this comment.

## Simple Component
Any component that returns a single child. Basically, any example you currently see in the docs that match one of the components in the parenthesis.

## Complex Component
Any component that contains multiple children with complex logic.

Does that help at all?

Reply:
@caseybaggz I think the reason we were asking was just to make sure that we had a chance to call out any components that exist in our BCs that we'd like to see in the Pando component library (_or at least know that we should plan to build something ourselves, if need be_).

I'm not sure I have anything specifically in mind right now, but if we had a list of what you all planned to included in the library (_even a loose list_ 😬 ), we could do a quick scan of our BCs to see if there's something missing from the list that we'd like to propose adding.

Reply:
Thanks Casey (:

Reply:
After reviewing [the docs](https://design.pluralsight.com/docs/learn/learn-pando/migration) detailing which Classic components will be recreated in Pando, the Search/Browse team would like to seek more context on why some components didn’t make the cut (ie. Drawer, Star Rating, and the heavily used Carousel).  To note, my team has already cutover to Headless Styles for all the Classic components that have replacements.  We had to continue using Classic for the missing components since we couldn’t find comparable functionality with third party libraries.  And as of this doc, it seems third party libraries are no longer recommended anyway.  So we would like to know what you recommend for users of these components when using HS or this new library.

Reply:
@GianaS thanks for your perspective and I appreciate you. If it helps - all I'll say since you mentioned a Carousel twice is that we are currently working with design/product to eliminate the carousel pattern because it is not a11y and no longer meets the standard we have as a company. But this conversation is outside of the scope of the RFC and I want to nip it in the bud now.

Regarding recommendations, we can always do that in our help channel but ultimately, Skills will have to create their own "extended library" at some point in time. This is not a new idea, Flow has done it to Classic almost immediately after being acquired years ago - it's just new for the Skills platform to not be catered to which I ultimately think causes the culture shock (just my 3rd perspective opinion).

There is reason to move off of specific Classic components due to a11y reason and just overall bad component patterns that were used. I know it may not be ideal for your team's use case right now, but daily we are making sure we are doing the best things we can do as the core team for long-term success of *all* the product teams which means that Pando is just never going to be Classic or repeat it's same patterns for these reasons, so I would encourage you to not think about comparison of the two which is just never going to make sense because it's comparing a carrot to an banana.

I want to end making sure you are taking away the right point though. No we aren't supporting those components now due to reasons and a system we are using internally as a team but that doesn't mean things won't change in the future. We originally didn't want a component library and look where we are today? 😉